cluster/images/hyperkube/master.json

```diff
@@ -23,6 +23,7 @@
               "apiserver",
               "--portal-net=10.0.0.1/24",
               "--address=127.0.0.1",
+              "--read-write-port=8080",
```

Comment:
That's exactly what I'm arguing :) I think that SSL should not be a hard requirement. There are setups (dev, getting started) where I see no reason to force people to generates certs etc. I think that having a cluster without SSL will make it easier to play with the setup, move to other providers, start it wherever and in general should improve perception of how complicated kubernetes setup is.

Assuming that we agree SSL should not be required, then we should work correctly in a cluster which is not using it. One example is a [single node docker based cluster](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/docker.md) which @brendandburns wrote. Currently pod<->apiserver communication doesn't work there. Adding certificates distribution would make the guide ~twice as long, so I don't think it's a good solution.